.github/workflows/code-engine-cleanup.yml

```diff
@@ -30,22 +42,29 @@ on:
 jobs:
   clean:
     runs-on: ubuntu-latest
-    env:
-      PR_TAG: pr-${{ github.event.pull_request.number }}
 
     steps:
+      - name: Set preview tag using input
+        if: "${{ inputs.preview_tag != '' }}"
+        run: echo "PREVIEW_TAG=${{ inputs.preview_tag }}" >> $GITHUB_ENV
+
+      - name: Set preview tag using PR number
+        if: "${{ inputs.preview_tag == '' }}"
+        run: echo "PREVIEW_TAG=pr-${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+
       - name: Create APP_NAME and DOCKER_IMAGE_TAG env variables
         run: |
-          echo "APP_NAME=${{ inputs.docker_image_name }}-${{ env.PR_TAG }}" >> $GITHUB_ENV
-          echo "DOCKER_IMAGE_TAG=icr.io/${{ inputs.container_registry_namespace }}/${{ inputs.docker_image_name }}:${{ env.PR_TAG }}" >> $GITHUB_ENV
+          echo "APP_NAME=${{ inputs.docker_image_name }}-${{ env.PREVIEW_TAG }}" >> $GITHUB_ENV
+          echo "DOCKER_IMAGE_TAG=icr.io/${{ inputs.container_registry_namespace }}/${{ inputs.docker_image_name }}:${{ env.PREVIEW_TAG }}" >> $GITHUB_ENV
 
       - name: GH deployment status (delete)
         id: gh_deployment
         uses: bobheadxi/deployments@v1
+        if: "${{ inputs.delete_env == 'true' }}"
```

Comment:
Related with my previous comment. In which cases are we going to want to delete an `env` if for PRs the environment will be always `Previews`?

Reply:
we probably wont need to delete an environment that is why default if `false` but i kept this step just in case soem future use care comes up that requires deleting an environment.